plugins/org.eclipse.glsp.api/src/org/eclipse/glsp/api/supplier/ContextActionsProviderSupplier.java

```diff
@@ -0,0 +1,32 @@
+/********************************************************************************
+ * Copyright (c) 2020 EclipseSource and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * https://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+package org.eclipse.glsp.api.supplier;
+
+import java.util.Collections;
+import java.util.Set;
+import java.util.function.Supplier;
+
+import org.eclipse.glsp.api.provider.ContextActionsProvider;
+
+public interface ContextActionsProviderSupplier extends Supplier<Set<ContextActionsProvider>> {
+   final class NullImpl implements ContextActionsProviderSupplier {
```

Comment:
I'm a bit confused by the term supplier in contrast to provider. Especially the combination "ProviderSupplier" doesn't make it clearer :)

Reply:
see https://github.com/eclipse-glsp/glsp-server/pull/46#discussion_r380065570
Sidenote: before the refactoring this would have been `ContextActionsProviderProvider` :sweat_smile: